pyiceberg/io/pyarrow.py

```diff
@@ -1122,12 +1121,12 @@ def project_table(
     return result
 
 
-def to_requested_schema(requested_schema: Schema, file_schema: Schema, table: pa.Table) -> pa.Table:
-    struct_array = visit_with_partner(requested_schema, table, ArrowProjectionVisitor(file_schema), ArrowAccessor(file_schema))
+def to_requested_schema(table: pa.Table, from_schema: Schema, to_schema: Schema) -> pa.Table:
```

Comment:
I refactored the helper method

Reply:
i can pull this refactor into a separate PR if it helps with review

Reply:
This is a public method, so we're breaking the API here. Not sure if a refactor justifies the the breaking change. Also, The `file_Schema` and `requested_schema` are more informative to me.

Reply:
I didn't realize it's a public API. I reverted the refactor